src/main/java/uk/gov/hmcts/reform/sscs/helpers/CaseFlagMappingServiceHelper.java

```diff
@@ -0,0 +1,106 @@
+package uk.gov.hmcts.reform.sscs.helpers;
+
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Service
+@SuppressWarnings("PMD")
+public class CaseFlagMappingServiceHelper {
+
+    private final String[] mappingValues =
+        {
+        "disabledAccess",
+        "signLanguageType",
+        "hearingLoop",
+        "isConfidentialCase",
+        "dwpUCB",
+        "dwpPHME",
+        "urgentCase",
+        "adjournCaseInterpreterLanguage"
+        };
+
+    private final Map<String, Object> caseFlags;
+
+    public CaseFlagMappingServiceHelper() {
+        this.caseFlags = new HashMap<>();
+    }
+
+    public Map<String, Object> mapCaseFlags(SscsCaseData caseData) {
+        mapSignLanguageType(caseData);
+        mapArrangements(caseData);
+        confidentialCase(caseData);
+        dwp(caseData);
+        urgentCase(caseData);
+        adjournCaseInterpreterLanguage(caseData);
+        return caseFlags;
+    }
+
+    private void mapSignLanguageType(SscsCaseData caseData) {
+        var signLanguageType = caseData.getAppeal().getHearingOptions().getSignLanguageType();
```

Comment:
Just spotted this sorry Zikrur, we can't guarantee the data in the nested objects is nonNull so this code could result in a NPE - you can wrap this in `Optional.ofNullable(caseData.getAppeal()..etc).ifPresent(e -> doStuff)`.

Reply:
There is a null check on it after that.